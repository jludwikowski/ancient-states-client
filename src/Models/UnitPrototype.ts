import {Entity} from './Entity';
import {UnitType} from './UnitType';

export default interface UnitPrototype extends Entity {
    baseTrainTime: number,
    baseDamage: number,
    activeTurns: number[],
    baseHealth: number,
    baseCost: number,
    type: UnitType,
    imageUrl: string
}
