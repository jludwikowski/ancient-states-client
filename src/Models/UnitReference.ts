export default interface UnitReference {
    unitId: number,
    number: number
}
