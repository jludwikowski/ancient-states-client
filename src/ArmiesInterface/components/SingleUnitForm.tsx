import React from "react";
import UnitPrototype from "../../Models/UnitPrototype";
import UnitReference from "../../Models/UnitReference";

interface Props {
    unit: UnitReference,
    unitPrototypes: UnitPrototype[],
    getUnit(event:React.ChangeEvent<HTMLInputElement>): void,
}

export const SingleUnitForm: React.FunctionComponent<Props> = props => (
    <div>
        <div>Unit Name: {props.unitPrototypes[props.unit.baseUnit].name}</div>
        <div>Level: {props.unit.level}</div>
        <div>Number of troops to take: {props.unit.number}</div>
        <input data-base={props.unit.baseUnit}
               data-level={props.unit.level}
               type="number"
               className="form-control"
               placeholder="0"
               name="number"
               onChange={ props.getUnit }
        />
    </div>
)
