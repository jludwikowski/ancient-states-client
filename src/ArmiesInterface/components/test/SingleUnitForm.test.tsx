import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import UnitPrototype from "../../../Models/UnitPrototype";
import UnitReference from "../../../Models/UnitReference";
import { SingleUnitForm } from '../SingleUnitForm';

describe('SingleUnitForm', () => {

    const unit:UnitReference = {
                number: 9802,
                level: 1,
                baseUnit: 1,
    };
    const baseUnits:UnitPrototype[] = [{
        id: 1,
        name: "Persian Spearman",
        description: "Poorly armed and armor eastern warrior",
        baseTrainTime: 50,
        baseDamage: 8,
        baseHealth: 30,
        baseCost: 40,
        activeTurns: [6, 7, 8, 9, 10, 11, 12],
        type: "Melee",
        imageUrl: "images/icons/icons8-military-base-100.png"
    }];

    const getUnit = (event:React.ChangeEvent<HTMLInputElement>) => {
        console.log('On onDismiss');
    };

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<SingleUnitForm unit={unit} unitPrototypes={baseUnits} getUnit={getUnit} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Should match snapshot', () => {
        const component = shallow(<SingleUnitForm unit={unit} unitPrototypes={baseUnits} getUnit={getUnit} />);
        expect(component).toMatchSnapshot();
    });
});
