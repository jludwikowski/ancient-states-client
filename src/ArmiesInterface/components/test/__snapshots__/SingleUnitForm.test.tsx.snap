// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleUnitForm Should match snapshot 1`] = `
<div>
  <div>
    Unit Name: 
    Persian Spearman
  </div>
  <div>
    Level: 
    1
  </div>
  <div>
    Number of troops to take: 
    9802
  </div>
  <input
    className="form-control"
    data-base={1}
    data-level={1}
    name="number"
    onChange={[Function]}
    placeholder="0"
    type="number"
  />
</div>
`;
